### PR TITLE
Add logged condition

### PR DESCRIPTION
**Pull request by idanarye**
Note that unlike `LoggedRLock`, no one is owning the condition while the waiters wait for it - so there is no point in logging the "number of owners".

### CODE REVIEW

File: easypy/concurrency.py [hunk 1]
```diff
@@ -1484,3 +1484,98 @@ def collect_and_call_once(self, param, func):
...
+            _logger.debug('%s - waiting to be acquired for ' + msg, self, *args)
+        try:
+            yield
+        finally:
+            self._cond.release()
+
+    @contextmanager
+    def notifying_all(self, msg, *args):
+        """
+        Acquire the condition lock for the context, and notify all all waiters afterward.
```
Comment (koreno) [inline easypy/concurrency.py]: all all

File: easypy/concurrency.py [hunk 2]
```diff
@@ -1484,3 +1484,98 @@ def collect_and_call_once(self, param, func):
...
+        Users should run the command that triggers the conditions inside this context manager.
+        """
+        with self._acquired_for('performing a %s notifying all waiters' % msg, *args):
+            yield
+            _logger.debug('%s - performed: ' + msg, self, *args)
+            self._cond.notifyAll()
+
+    def wait_for(self, pred, msg, *args, timeout=None):
+        """
+        Wait for a a predicate. Only check it when notified.
```
Comment (koreno) [inline easypy/concurrency.py]: `a a`?...

File: tests/test_concurrency.py [hunk 3]
```diff
@@ -257,3 +259,75 @@ def test_multiexception_api():
...
+    progress = 0
+    executed = []
+
+    def wait_for_progress_to(progress_to):
+        cond.wait_for(lambda: progress_to <= progress, 'progress to %s', progress_to)
+        executed.append(progress_to)
+
+    with concurrent(wait_for_progress_to, 10), concurrent(wait_for_progress_to, 20), concurrent(wait_for_progress_to, 30):
+        with patch("easypy.concurrency._logger") as _logger:
+            sleep(1)
```
Comment (koreno) [inline tests/test_concurrency.py]: (i try to make the sleeps as small as reasonably possible os that running pytest is quick)

File: tests/test_concurrency.py [hunk 4]
```diff
@@ -257,3 +259,75 @@ def test_multiexception_api():
...
+
+        assert sum(c == call("%s - waiting for progress to %s", cond, 10) for c in _logger.debug.call_args_list) > 3
+        assert sum(c == call("%s - waiting for progress to %s", cond, 20) for c in _logger.debug.call_args_list) > 3
+        assert sum(c == call("%s - waiting for progress to %s", cond, 30) for c in _logger.debug.call_args_list) > 3
+        assert executed == []
+
+        with patch("easypy.concurrency._logger") as _logger:
```
Comment (koreno) [inline tests/test_concurrency.py]: why not patch it for the entire test?

File: easypy/concurrency.py (hunk 2 above)
Comment (idanarye) [reply easypy/concurrency.py]: So nice I wrote it twice

File: tests/test_concurrency.py (hunk 4 above)
Comment (idanarye) [reply tests/test_concurrency.py]: Because I run different checks on each one I patch, so it's easier to not have the calls mixed.

**Pull request merged**